components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -0,0 +1,124 @@
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Define aerosol optics fields computed by this process.
+  auto nondim = Units::nondimensional();
+  FieldLayout scalar3d_swband_layout { {COL, SWBND, LEV}, {ncol_, nswbands_, nlev_} };
+  FieldLayout scalar3d_lwband_layout { {COL, LWBND, LEV}, {ncol_, nlwbands_, nlev_} };
+
+  // shortwave aerosol scattering asymmetry parameter [-]
+  add_field<Computed>("aero_g_sw",   scalar3d_swband_layout, nondim, grid_name);
+  // shortwave aerosol single-scattering albedo [-]
+  add_field<Computed>("aero_ssa_sw", scalar3d_swband_layout, nondim, grid_name);
+  // shortwave aerosol optical depth [-]
+  add_field<Computed>("aero_tau_sw", scalar3d_swband_layout, nondim, grid_name);
+  // longwave aerosol optical depth [-]
+  add_field<Computed>("aero_tau_lw", scalar3d_lwband_layout, nondim, grid_name);
+
+  // FIXME: this field doesn't belong here, but this is a convenient place to
+  // FIXME: put it for now.
+  // number mixing ratio for CCN
+  using Spack      = ekat::Pack<Real,SCREAM_SMALL_PACK_SIZE>;
+  using Pack       = ekat::Pack<Real,Spack::n>;
+  constexpr int ps = Pack::n;
+  FieldLayout scalar3d_layout_mid { {COL, LEV}, {ncol_, nlev_} };
+  add_field<Computed>("nccn", scalar3d_layout_mid, 1/kg, grid_name, ps);
+
+  logger.trace("leaving MAMOptics::set_grids");
+}
+
+void MAMOptics::initialize_impl(const RunType run_type) {
+}
+
+void MAMOptics::run_impl(const double dt) {
+
+  const auto policy = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(ncol_, nlev_);
+
+  // get the aerosol optics fields
+  auto aero_g_sw   = get_field_out("aero_g_sw");
+  auto aero_ssa_sw = get_field_out("aero_ssa_sw");
+  auto aero_tau_sw = get_field_out("aero_tau_sw");
+  auto aero_tau_lw = get_field_out("aero_tau_lw");
+
+  auto aero_nccn   = get_field_out("nccn"); // FIXME: get rid of this
+
+  // Compute optical properties on all local columns.
+  // (Strictly speaking, we don't need this parallel_for here yet, but we leave
+  //  it in anticipation of column-specific aerosol optics to come.)
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const ThreadTeam& team) {
+    const Int icol = team.league_rank(); // column index
+
+    auto g_sw = ekat::subview(aero_g_sw.get_view<Real***>(), icol);
+    auto ssa_sw = ekat::subview(aero_ssa_sw.get_view<Real***>(), icol);
+    auto tau_sw = ekat::subview(aero_tau_sw.get_view<Real***>(), icol);
+    auto tau_lw = ekat::subview(aero_tau_lw.get_view<Real***>(), icol);
+
+    // populate these fields with reasonable representative values
+    Kokkos::deep_copy(g_sw, 0.5);
+    Kokkos::deep_copy(ssa_sw, 0.7);
+    Kokkos::deep_copy(tau_sw, 0.0);
+    Kokkos::deep_copy(tau_lw, 0.0);
+
+    // FIXME: Get rid of this
+    auto nccn = ekat::subview(aero_nccn.get_view<Real**>(), icol);
+    Kokkos::deep_copy(nccn, 50.0);
+  });
+
+  /*
+  // update the timestamps for the fields
```

Comment:
I know this is commented, but just for clarity: the base class already does this _for all computed fields_. I'm not 100% sure that's the right thing to do, and we may change it in the future, but so far that's how it works.

Reply:
Thanks for the tip!